src/main/java/build/buf/protovalidate/internal/celext/CustomOverload.java

```diff
@@ -510,7 +561,9 @@ private static boolean validateHostname(String host) {
     }
     String s = Ascii.toLowerCase(host.endsWith(".") ? host.substring(0, host.length() - 1) : host);
     Iterable<String> parts = Splitter.on('.').split(s);
+    boolean allDigits = false;
```

Comment:
Upstream Go impl also picked up additional validation for hosts to ensure the last portion isn't all numbers.